src/sagas/client/clientSaga.ts

```
@@ -1,20 +1,53 @@
 import { all, takeLatest, call, put } from 'redux-saga/effects';
 import { AxiosResponse } from 'axios';
+import { get, find } from 'lodash';
 
-import { ClientActionTypes, FetchDataEntryPayload } from '../../reducers/client/clientTypes';
+import { ClientActionTypes, FetchDataEntryPayload, UpdateDataEntryPayload } from '../../reducers/client/clientTypes';
 import ClientService from './clientService';
 import { APIResponse, getAPIErrorMessage } from '../../utils/apiUtils';
+import { POSITIONS } from '../../enums/client';
 
 export default class ClientSaga {
   public static *fetchDataEntry({ payload }: { payload: FetchDataEntryPayload }) {
     try {
       const { clientId, tabName, tagName } = payload;
-      const response: AxiosResponse<APIResponse> = yield call(ClientService.fetchDataEntry);
+      const tabNameValue = get(find(POSITIONS, ({ slug }) => slug === tabName), 'value');
+      if (tabNameValue) {
+        const response: AxiosResponse<APIResponse> = yield call(ClientService.fetchDataEntry, clientId, tabNameValue);
+        if (response.status === 200 && response.data.success) {
+          yield put({
+            type: ClientActionTypes.FETCH_DATA_ENTRY_SUCCESS,
+            payload: {
+              clientId,
+              tagName,
+              tabName,
+              dataEntry: response.data.data,
+            },
+          });
+        }
```

What happened if the Axios call is failed? We should throw an error here

---

@dinhnguyentrunghoa If the called API has failed, Axios will execute the rejected case and the `catch` block will handle this situation.